UnitTests/CommonTestUtils/MockExecutable.cs

```diff
@@ -133,6 +134,18 @@ public Task<int> WaitForExitAsync()
             {
                 if (_exitCode.HasValue)
                 {
+                    if (_throwOnErrorOutput)
+                    {
+#pragma warning disable VSTHRD103 // Call async methods when in an async method
+                        string errorOutput = StandardError.ReadToEnd().Trim();
+#pragma warning restore VSTHRD103 // Call async methods when in an async method
+                        if (_exitCode.Value != 0 || errorOutput.Length > 0)
+                        {
+                            ExternalOperationException ex = new(exitCode: _exitCode, innerException: new Exception(errorOutput));
+                            return Task.FromException<int>(ex);
+                        }
+                    }
+
```

Comment:
@mstv this is a "poor man" integration tests for https://github.com/gitextensions/gitextensions/pull/9056. This implementation must match the "prod" functionality.